tests/utils.c

```diff
@@ -426,17 +441,75 @@ int test_memio_copy_message(const struct test_memio_ctx *ctx, int client,
     if (msg_pos < 0 || msg_pos >= msg_count) {
         return -1;
     }
-    if (*out_sz < msg_sizes[msg_pos]) {
-        return -1;
-    }
     for (i = 0; i < msg_pos; i++) {
         buff += msg_sizes[i];
     }
-    XMEMCPY(out, buff, (size_t)msg_sizes[msg_pos]);
+    *out = (const char*)buff;
     *out_sz = msg_sizes[msg_pos];
     return 0;
 }
 
+int test_memio_move_message(struct test_memio_ctx *ctx, int client,
```

Comment:
can you please:

1. add a test_memio_inject_message_at that takes an index and inject a message inside the buffer, example: buff = [ AAA, BBBB, CCCC], test_memio_inject_message_at(XXXX, 2) transforms the buffer in = [AAA, BBBB, XXXX, CCCC].

2. then rewrite the "test_memio_move_message" as a test_memio_copy_message, test_memio_drop_message and test_memio_inject_message_at

Reply:
That will require either alloc'ing memory or using another full `TEST_MEMIO_BUF_SZ` on the stack which is already 64KB. `test_memio_ctx` is already large enough. I'm not sure how much we run `unit.test` on smaller targets but I think this might hit some memory limits. It would probably be beneficial to move `memio` to heap allocation at some point.

Reply:
can you rewrite so that you first reserve the space for the message at destination, copy the message from the origin and then invoke drop_message on the message in the original position.

It would be great if we maintain test functions very simple even if they are not optimized imo.
As it is, `test_memio_move_message` is hard to read.

Reply:
Reserving spaces requires manipulating the IO buffer. I don't think that is possible as finding the correct message requires the buffer to match the message sizes. I would argue that all of the memio functions are hard to read since they need to respect the message sizes. Ironically, I think switching to heap allocated buffers would make it easier to read and manipulate.